src/step/assemble_tangent_operator.hpp

```diff
@@ -19,7 +19,10 @@ inline void AssembleTangentOperator(Solver& solver, State& state) {
     const auto col_idx_size = Kokkos::View<int*>::shmem_size(max_row_entries);
     auto sparse_matrix_policy = Kokkos::TeamPolicy<>(static_cast<int>(num_nodes), Kokkos::AUTO());
 
-    sparse_matrix_policy.set_scratch_size(1, Kokkos::PerThread(row_data_size + col_idx_size));
+    // TODO: Hack to avoid the segfault. Why do we need to increase the scratch size here?
```

Comment:
Yes

Reply:
@ddement Following is the context for this: 

> I added an r-test for the [precession problem](https://github.com/deslaughter/ottr/blob/da3c097380a832007b4439da1d56082855e3aaee/tests/rigid_body_test.rs#L18), however I was seeing a segfault from a scratch memory size allocation in AssembleTangentOperator. With a quick hack of just increasing that, I see the sparse solver now fails to factorize the system: "KLU2 numeric factorization failed". Would appreciate any thoughts/help to resolve this issue.

Reply:
I pushed a more permanent fix.  I don't know why the memory pool was running out of memory in this specific instance - it was fine for all of our beams based problems with no special logic.  I might look into it more later.  The fix is to use stack memory for the scrach memory, since we only need space for a maximum of 6 entries. 

We still have problem with the linear system solve failing.  This is either going to be because our matrices are populated incorrectly from our residual/matrix calculations or because the values are computed incorrectly in the first place.  @deslaughter @faisal-bhuiyan Do you want me to try to debug this?

Reply:
I never went too deep into debugging the system matrices i.e. if the values are correct or not-- let me take a stab at this and I will post some update here.

Reply:
@ddement Yes, I'd like for you to debug it. I've got this test implemented in Rust, so I can give you the system matrices for checking. Let me know what you need.